infrastructure/src/main/java/org/corfudb/infrastructure/LogReplicationServer.java

```diff
@@ -79,34 +85,80 @@ public void shutdown() {
     private void handleLogReplicationEntry(CorfuPayloadMsg<LogReplicationEntry> msg, ChannelHandlerContext ctx, IServerRouter r) {
         log.info("Log Replication Entry received by Server.");
 
-        LogReplicationEntry ack = sinkManager.receive(msg.getPayload());
+        if (isLeader(msg, r)) {
```

Comment:
We don't want to process any incoming message, if we are not the leader. If we received a message it's because we are believed to be the leader by the source cluster, however, we aren't.  For this reason we reply that we've lost leadership as a way to trigger the source to re-verify leader on the remote.